server/src/main/java/org/opensearch/common/geo/parsers/GeoWKTParser.java

```diff
@@ -265,21 +270,23 @@ private static MultiLineStringBuilder parseMultiLine(StreamTokenizer stream, fin
         return builder;
     }
 
-    private static PolygonBuilder parsePolygon(StreamTokenizer stream, final boolean ignoreZValue, final boolean coerce)
-            throws IOException, OpenSearchParseException {
+    private static PolygonBuilder parsePolygon(StreamTokenizer stream, final boolean ignoreZValue, final boolean coerce) throws IOException,
+        OpenSearchParseException {
         if (nextEmptyOrOpen(stream).equals(EMPTY)) {
             return null;
         }
-        PolygonBuilder builder = new PolygonBuilder(parseLinearRing(stream, ignoreZValue, coerce),
-            AbstractShapeGeometryFieldMapper.Defaults.ORIENTATION.value());
+        PolygonBuilder builder = new PolygonBuilder(
+            parseLinearRing(stream, ignoreZValue, coerce),
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the resource returned by the *parseLinearRing* method in a try-finally block. Currently, there are execution paths that do not contain closure statements, for example, when Explicit.value throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).